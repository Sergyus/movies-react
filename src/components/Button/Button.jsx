import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './Button.scss';

const Button = (props) => {
  const { children, type, handleClick } = props;
  return (
    <button
      aria-label={children}
      type="button"
      onClick={handleClick}
      className={cn(styles.btn, styles[type])}
    >
      { children }
    </button>
  );
};

Button.propTypes = {
  type: PropTypes.string.isRequired,
  handleClick: PropTypes.func,
  children: PropTypes.string.isRequired,
};

Button.defaultProps = {
  handleClick: () => {},
};

export default Button;
