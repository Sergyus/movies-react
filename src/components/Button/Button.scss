@import "../../styles/variables.scss";

.btn {
  font-size: 1rem;
  cursor: pointer;
  color: white;
  border-radius: 3px;
  width: 100px;
  max-height: 35px;
  padding: 7px;
  font-family: "Helvetica Neue", Arial, Helvetica, sans-serif;
  outline: none;
  transition: transform 0.2s;
}

.ghost {
  border: 1px solid white;
  background-color: transparent;

  &:hover {
    transform: scale(1.05);
  }
}

.primary {
  border: none;
  background-color: $blue-color;

  &:hover {
    transform: scale(1.05);
  }
}
