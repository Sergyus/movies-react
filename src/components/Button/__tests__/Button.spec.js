import React from 'react';
import Button from '..';
import { create, act } from 'react-test-renderer';

describe('Button component renders correctly', () => {
  it('Primary button renders correctly', () => {
    const tree = create(<Button type="primary">Watch Now</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Ghost button renders correctly', () => {
    const tree = create(<Button type="ghost">View Info</Button>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Click button with handler works correctly', () => {
    const mockFunction = jest.fn();
    const tree = create(
      <Button handleClick={mockFunction} type="ghost"> View Info </Button>,
    );
    const button = tree.root.children[0];
    act(() => {
      button.props.onClick();
    });
    expect(mockFunction.mock.calls.length).toEqual(1);
  });

  it('Click button without handler works correctly', () => {
    const tree = create(
      <Button type="ghost">View Info</Button>,
    );
    const button = tree.root.children[0];
    const mockFunction = jest.fn(button.props.onClick);
    act(() => {
      mockFunction();
    });
    expect(mockFunction).toHaveBeenCalled();
  });
});
