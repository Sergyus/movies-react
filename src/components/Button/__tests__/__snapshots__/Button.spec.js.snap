// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly Ghost button renders correctly 1`] = `
<button
  aria-label="View Info"
  className="btn ghost"
  onClick={[Function]}
  type="button"
>
  View Info
</button>
`;

exports[`Button component renders correctly Primary button renders correctly 1`] = `
<button
  aria-label="Watch Now"
  className="btn primary"
  onClick={[Function]}
  type="button"
>
  Watch Now
</button>
`;
